src/azure-cli/azure/cli/command_modules/appconfig/feature.py

```diff
@@ -878,77 +983,22 @@ def __list_all_keyvalues(azconfig_client,
     # (?:\\\\)*  Matches any number of occurrences of two backslashes
     # ,          Matches a comma
     unescaped_comma_regex = re.compile(r'(?<!\\)(?:\\\\)*,')
-    if unescaped_comma_regex.search(feature):
+    if unescaped_comma_regex.search(key_filter):
         raise CLIError("Comma separated feature names are not supported. Please provide escaped string if your feature name contains comma. \nSee \"az appconfig feature list -h\" for correct usage.")
 
-    # Filtering keys on these patterns needs to happen on client side after getting all keys that match user specified pattern
-    # If user provides *abc or *abc* or * -> get all keys that match this
-    # pattern, then filter based on whether they are feature flags or not
-    all_keys_pattern = "*"
-    if feature.startswith("*") and feature != all_keys_pattern:
-        key = feature
-    else:
-        key = FeatureFlagConstants.FEATURE_FLAG_PREFIX + feature
-
-    # If user has specified fields, we still get all the fields and then
-    # filter what we need from the response.
-
     label = prep_label_filter_for_url_encoding(label)
 
     try:
-        configsetting_iterable = azconfig_client.list_configuration_settings(key_filter=key, label_filter=label)
+        configsetting_iterable = azconfig_client.list_configuration_settings(key_filter=key_filter, label_filter=label)
     except HttpResponseError as exception:
         raise CLIError('Failed to read feature flag(s) that match the specified feature and label. ' + str(exception))
 
     try:
         retrieved_kv = [convert_configurationsetting_to_keyvalue(x) for x in configsetting_iterable]
-        if key != feature:
-            valid_features = []
-            for kv in retrieved_kv:
-                if kv.content_type == FeatureFlagConstants.FEATURE_FLAG_CONTENT_TYPE:
-                    valid_features.append(kv)
-            return valid_features
-
-        return __custom_key_filtering(retrieved_kv=retrieved_kv, user_key_filter=feature)
+        valid_features = []
+        for kv in retrieved_kv:
+            if kv.content_type == FeatureFlagConstants.FEATURE_FLAG_CONTENT_TYPE:
+                valid_features.append(kv)
+        return valid_features
     except Exception as exception:
         raise CLIError(str(exception))
-
-
-def __custom_key_filtering(retrieved_kv, user_key_filter):
```

Comment:
Why don't we need this logic anymore?

Reply:
We stopped supporting this filtering logic on the service side too. This is just cleanup of functionality that currently doesn't work anyway.

Reply:
Got it, thanks~